snuba/clusters/cluster.py

```diff
@@ -88,7 +88,18 @@ def __init__(self, storage_sets: Set[str]):
         self.__storage_sets = storage_sets
 
     def get_storage_set_keys(self) -> Set[StorageSetKey]:
-        return {StorageSetKey(storage_set) for storage_set in self.__storage_sets}
+        storage_set_keys = set()
+
+        for storage_set in self.__storage_sets:
+            try:
+                storage_set_keys.add(StorageSetKey(storage_set))
+            except ValueError:
```

Comment:
There is a better way to check whether a key or a value is present in an Enum. catching a very generic exception would be a smell because, while unlikely, that could be raised for other reasons.
https://docs.python.org/3/library/enum.html#iteration